src/impl/keyence_high_speed_tcp.cpp

```diff
@@ -0,0 +1,197 @@
+#include "keyence/impl/keyence_high_speed_tcp.h"
+#include "boost/thread.hpp"
+#include "boost/atomic.hpp"
+
+#include "keyence/impl/messages/prepare_high_speed.h"
+#include "keyence/impl/messages/start_high_speed.h"
+#include "keyence/impl/messages/stop_high_speed.h"
+#include "keyence/impl/keyence_exception.h"
+#include "keyence/impl/ljv7_rawdata.h"
+
+#include "libsocket/epoll.hpp"
+
+#define KEYENCE_HIGHSPEED_NOTIFY_MESSAGE_SIZE 8
+
+namespace keyence
+{
+
+static void worker_main(boost::atomic_flag& flag,
+                        libsocket::inet_stream& sock,
+                        keyence::HighSpeedTcpClient::callback_type cb,
+                        uint16_t profile_size)
+{
+  using libsocket::inet_stream;
+  // prepare buffers
+  const size_t msg_length = 24 + (profile_size * 5 / 2) + 4;
+  std::vector<unsigned char> byte_buffer (msg_length);
+  std::vector<int32_t> points_buffer (profile_size);
+  // keep track of the 'message state'
+  size_t received = 0;
+
+  libsocket::epollset<inet_stream> poll;
+  poll.add_fd(sock, LIBSOCKET_READ);
+
+  while (flag.test_and_set())
+  {
+    libsocket::epollset<inet_stream>::ready_socks ready = poll.wait(100);
+    if (!ready.first.empty())
+    {
+      // read into buffer
+      ssize_t n = sock.rcv(byte_buffer.data() + received, msg_length - received);
+      if (n <= 0)
+      {
+        // Socket has been closed
+        std::cerr << "High speed socket closed\n";
+        break;
+      }
+
+      received += n;
+
+      if (received == msg_length)
```

Comment:
This `if/else` seems fragile.  Does it rely on the socket read returning "full packets".  What if the socket is a partial read of a valid profile, but unluckily the socket bytes received is equal to `KEYENCE_HIGHSPEED_NOTIFY_MESSAGE_SIZE`